services/wallet/onramp/on_ramp.go

```diff
@@ -0,0 +1,119 @@
+package onramp
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type DataSourceType int
+
+const (
+	DataSourceHTTP DataSourceType = iota + 1
+	DataSourceStatic
+)
+
+type CryptoOnRamp struct {
+	Name              string            `json:"name"`
+	Description       string            `json:"description"`
+	Fees              string            `json:"fees"`
+	LogoURL           string            `json:"logoUrl"`
+	SiteURL           string            `json:"siteUrl"`
+	RecurrentSiteURL  string            `json:"recurrentSiteUrl"`
+	Hostname          string            `json:"hostname"`
+	Params            map[string]string `json:"params"` // TODO implement params in JSON and parsing status-mobile
```

Comment:
I know the comment was not added by you, but if it's still relevant, can we move this to a GitHub issue? It will be easier to track.

Reply:
Same as this I think... https://github.com/status-im/status-go/pull/5231#discussion_r1617208925
This needs some cleanup/rework probably, but I want to change as little as possible until there's a real need.